apiserver/observer/request_notifier.go

```diff
@@ -71,13 +85,23 @@ func (n *RequestObserver) isAgent(entity names.Tag) bool {
 }
 
 // Login implements Observer.
-func (n *RequestObserver) Login(entity names.Tag, model names.ModelTag, fromController bool, _ string) {
+func (n *RequestObserver) Login(entity names.Tag, model names.ModelTag, fromController bool, userData string) {
 	n.state.tag = entity.String()
-	// Don't log connections from the controller to the model.
-	if n.isAgent(entity) && !fromController {
+	n.state.fromController = fromController
+	if n.isAgent(entity) {
 		n.state.agent = true
 		n.state.model = model.Id()
-		n.connLogger.Infof("agent login: %s for %s", n.state.tag, n.state.model)
+		// Don't log connections from the controller to the model.
+		if !n.state.fromController {
+			n.connLogger.Infof("agent login: %s for %s", n.state.tag, n.state.model)
+		}
+		n.hub.Publish(apiserver.ConnectTopic, apiserver.APIConnection{
```

Comment:
Is this going to have the same problem I had with the peergrouper? Or will you be able to guarantee that subscribers to ConnectTopic will be subscribed before this?

Reply:
Yes it does, and fixed a similar way. This is coming in the presence-worker branch.